salt/orchestrate/aws/rds.sls

```diff
@@ -67,16 +67,23 @@ create_{{ ENVIRONMENT }}_{{ dbconfig.name }}_rds_store:
     - require:
         - boto_rds: create_{{ ENVIRONMENT }}_rds_db_subnet_group
 
+{% set mount_point = '{}-{}-{}'.format(dbconfig.engine, ENVIRONMENT, dbconfig.name) %}
 configure_vault_postgresql_{{ dbconfig.name }}_backend:
   vault.secret_backend_enabled:
-    - backend_type: postgresql
-    - description: Backend to create dynamic PostGreSQL credentials for {{ ENVIRONMENT }}
-    - mount_point: postgresql-{{ ENVIRONMENT }}-{{ dbconfig.name }}
+    - backend_type: database
+    - description: Backend to create dynamic {{ dbconfig.engine }} credentials for {{ ENVIRONMENT }}
+    - mount_point: {{ mount_point }}
     - ttl_max: {{ SIX_MONTHS }}
     - ttl_default: {{ SIX_MONTHS }}
     - lease_max: {{ SIX_MONTHS }}
     - lease_default: {{ SIX_MONTHS }}
+    - connection_config_path: 
     - connection_config:
+        plugin_name: {{ dbconfig.vault_plugin }}
+        {% if dbconfig.engine == 'postgres' %}
         connection_url: "postgresql://{{ master_user }}:{{ master_pass }}@postgresql-{{ dbconfig.name }}.service.{{ ENVIRONMENT }}.consul:5432/{{ dbconfig.name }}"
+        {% else %}
+        connection_url: "{{ master_user }}:{{ master_pass }}@tcp(mysql.service.{{ ENVIRONMENT }}.consul:3306)/"
```

Comment:
Why did you leave out the {{ dbconfig.name }} at the end of this connection_url?

Reply:
Because it's not required for MySQL connections